File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string res="";
        for(auto x : strs){
            res += to_string(x.size()) + "#" + x;
        }
        return res;
}
        

    vector<string> decode(string s) {
        vector<string>res;
        int i=0,n=s.size();
        while(i<n){
            int j=i;
            while(s[j] != '#')
                j++;
            int length = stoi(s.substr(i, j-i));
            res.push_back(s.substr(j+1,length));
            i = j + 1 + length;
        }
        return res;
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = []
        n = len(temperatures)
        
        for i in range(n):
            for j in range(i,n):
                if temperatures[i] <  temperatures[j]:
                    res.append(j-i)
                    break
                elif j >= n-1:
                    res.append(0)

        return res
             